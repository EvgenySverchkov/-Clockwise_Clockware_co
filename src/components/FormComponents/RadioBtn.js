import React from "react";
import PropTypes from "prop-types";

const RadioBtn = ({ id, value, name, chngHandler, children }) => (
  <div className="form-check form-check-inline">
    <input
      className="form-check-input"
      type="radio"
      onChange={chngHandler || null}
      name={name}
      id={id}
      value={value}
    />
    {children}
  </div>
);

RadioBtn.propTypes = {
  id: PropTypes.node.isRequired,
  value: PropTypes.node.isRequired,
  name: PropTypes.string.isRequired,
  chngHandler: PropTypes.func,
  children: PropTypes.node,
};

export default RadioBtn;
