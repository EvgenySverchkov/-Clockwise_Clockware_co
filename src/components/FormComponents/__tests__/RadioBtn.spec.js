import React from 'react';
import {create} from "react-test-renderer";

import RadioBtn from "../RadioBtn";

it("<RadioBtn /> test which is used in forms", ()=>{
    const mockProps = {
        id: "1", 
        value: "Value", 
        name: "Name", 
        chngHandler: jest.fn()
    }
    const component = create(<RadioBtn {...mockProps}>
        <div>Some element</div>
    </RadioBtn>)
    expect(component.toJSON()).toMatchSnapshot();
});