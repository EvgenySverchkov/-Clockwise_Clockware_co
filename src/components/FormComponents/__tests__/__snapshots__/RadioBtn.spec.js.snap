// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioBtn /> test which is used in forms 1`] = `
<div
  className="form-check form-check-inline"
>
  <input
    className="form-check-input"
    id="1"
    name="Name"
    onChange={[MockFunction]}
    type="radio"
    value="Value"
  />
  <div>
    Some element
  </div>
</div>
`;
